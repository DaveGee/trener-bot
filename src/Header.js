import React from "react";
import { Auth } from 'aws-amplify'

import {
  AppBar,
  Toolbar,
  Typography,
  Button
} from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
  },
  title: {
    flexGrow: 1,
  },
}))

async function signOut() {
  await Auth.signOut()
    .then(data => console.log(data))
    .catch(err => console.log(err));
}

export default () => {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6" className={classes.title}>
            Trener Bot
      </Typography>
          <Button color="inherit" onClick={signOut}>Logout</Button>
        </Toolbar>
      </AppBar>
    </div>
  )
}