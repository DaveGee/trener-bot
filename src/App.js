import React, { useEffect, useState } from 'react'

import { 
  CssBaseline, 
  Container, 
  createMuiTheme, 
  ThemeProvider 
} from '@material-ui/core'

import Header from './Header'
import CardList from './CardList/CardList'

import Amplify, { Auth } from 'aws-amplify'
import awsconfig from './aws-exports'
import { withAuthenticator } from 'aws-amplify-react'

import teal from '@material-ui/core/colors/teal'
import deepOrange from '@material-ui/core/colors/deepOrange'

Amplify.configure(awsconfig)

const theme = createMuiTheme({
  palette: {
    primary: teal,
    secondary: deepOrange,
  },
})

function App() {
  const [user, setUser] = useState(null)

  useEffect(() => {
    const getUser = async () => {
      const user = await Auth.currentAuthenticatedUser()
      setUser(user)
    }
    getUser()
  }, [])

  return (
    <React.Fragment>
      <CssBaseline />
      <ThemeProvider theme={theme}>
        <Container fixed maxWidth="md">
          <Header />
          {user && <CardList owner={user} />}
        </Container>
      </ThemeProvider>
    </React.Fragment>
  )
}

const signUpConfig = {
  header: 'Choose an email and password',
  hideAllDefaults: true,
  defaultCountryCode: '41',
  signUpFields: [
    {
      label: 'Email',
      key: 'email',
      required: true,
      displayOrder: 1,
      type: 'string'
    },
    {
      label: 'Password',
      key: 'password',
      required: true,
      displayOrder: 2,
      type: 'password'
    },
    {
      label: 'Name',
      key: 'name',
      required: false,
      displayOrder: 3,
      type: 'string'
    },
  ]
};

export default withAuthenticator(App, {
  signUpConfig,
  usernameAttributes: 'email',
})
