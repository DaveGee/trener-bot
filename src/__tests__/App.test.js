import React from 'react'
import { render } from '@testing-library/react'
import App from '../App'
import Header from '../Header'

it('renders App', () => {
  const { asFragment } = render(<App />)
  expect(asFragment()).toMatchSnapshot()
})

it('renders Header', () => {
  const { asFragment } = render(<Header />)
  expect(asFragment()).toMatchSnapshot()
})
