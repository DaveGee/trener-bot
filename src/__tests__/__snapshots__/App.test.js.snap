// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<DocumentFragment>
  <div
    class="Section__container___3YYTG"
  >
    <div
      class="Form__formContainer___1GA3x"
    >
      <div
        class="Form__formSection___1PPvW"
      >
        <div
          class="Section__sectionBody___ihqqd"
        >
          Loading...
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders Header 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
        >
          Trener Bot
        </h6>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Logout
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
